package-system/zlib/build_zlib_linux.sh

```diff
@@ -6,6 +8,21 @@
 #
 #
 
+
+# Building zlib on ubuntu requires the following packages
+REQUIRED_DEV_PACKAGES="cmake ninja-build gcc"
+ALL_PACKAGES=`apt list | grep installed  2>/dev/null`
+for req_package in $REQUIRED_DEV_PACKAGES
+do
+    PACKAGE_COUNT=`echo $ALL_PACKAGES | grep $req_package | wc -l`
+    if [[ $PACKAGE_COUNT -eq 0 ]]; then
+        echo "Missing required package '${req_package}'. Install this package and try again."
+        exit 1
+    fi
+done
```

Comment:
The better way to run command substitution is to use `$()`
```suggestion
ALL_PACKAGES=$(apt list | grep installed  2>/dev/null)
for req_package in $REQUIRED_DEV_PACKAGES
do
    PACKAGE_COUNT=$(echo $ALL_PACKAGES | grep $req_package | wc -l)
    if [[ $PACKAGE_COUNT -eq 0 ]]; then
        echo "Missing required package '${req_package}'. Install this package and try again."
        exit 1
    fi
done
```

Reply:
This was copied from another existing script. I wish there was a better way to do this, but i couldn't find any way to do a list of installed packages without a grep.